Bitblaster code cleanup

Ovascos opened the pull request:
Clean up some bitblaster and BV solver code.
- remove some dead code from `TBitblaster` and `NodeBitblaster` (reference to SAT solver and model cache which was unused; removed duplicated model evaluation code from the bitblaster)
- Aligned variable naming and assertion style in `bitblast_strategies_template.h`
- Added some missing `const` and `override` and fixed some typos

Ovascos committed:
Update applyAtomBBStrategy to avoid ternary operator at every call.

Ovascos committed:
Aligned different functions in bitblast_strategies_template.h

Ovascos committed:
Bitblaster cleanup

- Removed dead code from bitblaster classes
- Moved collectModelValues to the bv_solver_bitblast_internal, to align with all other BV solvers.

Ovascos committed:
Fixed typos and code formatting in bitblaster code

Ovascos committed:
Bitblaster cleanup

- using reference to TheoryState instead of pointer
- added some missing destructor overrides

Ovascos committed:
clang-formatting

Ovascos committed:
Clear split between TBitblaster and Nodebitblaster

TBitblaster contains the abstract strategies only. NodeBitblaster (as
the user of the abstract strategies) takes care about everyting related
to state (incl. node handling, caching,...).

Ovascos committed:
Minor fixes

Ovascos committed:
minor fixes in theory_bv

Ovascos committed:
Added makeAtom to TBitblaster

Ovascos committed:
minor fixes

Ovascos committed:
Minor changes in bv_solver_bitblast

Removed d_propagate to use setting directly and fixed includes.

Ovascos committed:
Don't init proof code when proofs are disabled.

mpreiner commented on src/theory/bv/bitblast/bitblast_strategies_template.h:
```diff
@@ -463,7 +479,7 @@ void uDivModRec(NodeManager* nm,
                 unsigned rec_width)
 {
-  Assert(q.size() == 0 && r.size() == 0);
+  Assert(q.empty() && r.empty());
```
```suggestion
  Assert(q.empty());
  Assert(r.empty());
```

Ovascos committed:
Properly register boolean variables in eager mode

Instead of just collecting values for the variables, properly register
them as relevant terms.

Ovascos committed:
Encapsulate bitblaster and cnf stream code

Extract all code related to using the bitblaster and cnf stream from the
`BVSolverBitblast` into a new class to better separate concerns.

Ovascos committed:
BV rewriter code rework.
